Libraries/Vibration/Vibration.js

```diff
@@ -27,11 +27,31 @@ var Platform = require('Platform');
  */
 
 var Vibration = {
-  vibrate: function(duration: number = 400) {
+  vibrate: function(pattern: number | Array<number> = 400, repeat: number = -1) {
```

Comment:
This is just a straight copy of Android (http://developer.android.com/reference/android/os/Vibrator.html#vibrate(long[], int))

That said, I tend to agree.  I wonder if we should do just a boolean for repeat or not, and not allow for the direct indexing into the pattern for repeat.